src/test/java/com/spring/rest/dao/impl/CarCharacteristicsDaoImplTest.java

```diff
@@ -0,0 +1,97 @@
+package com.spring.rest.dao.impl;
+
+import com.spring.rest.dao.CarCharacteristicsDao;
+import com.spring.rest.model.CarCharacteristics;
+import com.spring.rest.model.CarMake;
+import com.spring.rest.model.CarModel;
+import com.spring.rest.model.FuelType;
+import com.spring.rest.model.Transmission;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.dao.EmptyResultDataAccessException;
+import java.util.Date;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@SpringTestConfiguration
+class CarCharacteristicsDaoImplTest {
+
+    CarCharacteristicsDao carCharacteristicsDao;
+
+    @Autowired
+    CarCharacteristicsDaoImplTest(CarCharacteristicsDao carCharacteristicsDao) {
+        this.carCharacteristicsDao = carCharacteristicsDao;
+    }
+
+    @Test
+    void getCarsCharacteristics() {
```

Comment:
It's incorrect.
It would be better to remember number of entities before deleting, then remove one of them and check that current number of available entities decreased by 1.